src/interpreter/ByteCode.h

```diff
@@ -356,22 +366,20 @@ class BinaryOperation : public ByteCode {
     const ByteCodeStackOffset* srcOffset() const { return m_srcOffset; }
     ByteCodeStackOffset dstOffset() const { return m_dstOffset; }
     void setDstOffset(ByteCodeStackOffset o) { m_dstOffset = o; }
-#if !defined(NDEBUG)
-    void dump(size_t pos)
-    {
-    }
-#endif
 
 protected:
     ByteCodeStackOffset m_srcOffset[2];
     ByteCodeStackOffset m_dstOffset;
 };
 
 #if !defined(NDEBUG)
-#define DEFINE_BINARY_BYTECODE_DUMP(name)                                                                                                             \
-    void dump(size_t pos)                                                                                                                             \
-    {                                                                                                                                                 \
-        printf(#name "src1: %" PRIu32 " src2: %" PRIu32 " dst: %" PRIu32, (uint32_t)m_srcOffset[0], (uint32_t)m_srcOffset[1], (uint32_t)m_dstOffset); \
+#define DEFINE_BINARY_BYTECODE_DUMP(name)                                                                                                       \
+                                                                                                                                                \
+    const std::string getName() const { return #name; }                                                                                         \
+                                                                                                                                                \
+    void printExtra(size_t pos)                                                                                                                 \
+    {                                                                                                                                           \
+        printf("src1: %" PRIu32 " src2: %" PRIu32 " dst: %" PRIu32, (uint32_t)m_srcOffset[0], (uint32_t)m_srcOffset[1], (uint32_t)m_dstOffset); \
     }
```

Comment:
Why did you separate `getName` and `printExtra` function?
How about merging these two functions into one function?

Reply:
I separated them to put them in different columns in the debug mode table.